pkg/controller/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -317,3 +321,44 @@ func (r *ReconcileOCSInitialization) Reconcile(request reconcile.Request) (recon
 
 	return reconcile.Result{}, err
 }
+
+// returns a ConfigMap with default settings for rook-ceph operator
+func newRookCephOperatorConfig(namespace string) *corev1.ConfigMap {
+	var defaultCSIToleration string = `
+- key: node.ocs.openshift.io/storage
+  operator: Equal
+  value: "true"
+  effect: NoSchedule`
+
+	config := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rookCephOperatorConfigName,
+			Namespace: namespace,
+		},
+	}
+	data := make(map[string]string)
+	data["CSI_PROVISIONER_NODE_AFFINITY"] = "cluster.ocs.openshift.io/openshift-storage="
+	data["CSI_PROVISIONER_TOLERATIONS"] = defaultCSIToleration
+	data["CSI_PLUGIN_TOLERATIONS"] = defaultCSIToleration
+	data["CSI_ENABLE_SNAPSHOTTER"] = "false"
```

Comment:
If these settings are never going to be reconciled again, we can only add settings that truly should be configured by the user. Specifically, whether to enable the snapshotter should not be set here. We should keep that as an env variable in the operator spec. Otherwise, the snapshotter will be disabled and will not be enabled in the future when we expect it to be enabled. It is still possible for the user to override the default, but we can't set a default where OCS needs to change the default in the future.

Reply:
Just to verify: I imagine ConfigMap settings override env vars in the Deployment for Rook-Ceph?

Reply:
correct, the CM settings override the env vars

Reply:
Makes sense. Will make the change.

Just to clear it out, even though we are creating this ConfigMap, I have retained the Env Vars in the deployment in CSV as a fall-back.

Meaning, if user wants to override anything, that can be done via CM. But, if user removes any of the defaults, the Env Var will be picked and applied to ensure that our recommended defaults are honoured.

If that is not desired, the setting key's value can be set to `nil` and we'll not fall-back.

Reply:
Fixed this.